### PR TITLE
provider/aws: Add docs for elasticache_cluster - cache_nodes

### LINKED ISSUE #2272: aws_elasticache_cluster doesn't document `cache_nodes`
It's tough to use `aws_elasticache_cluster` right now, since it doesn't expose the endpoint hostname for nodes in the cluster (maybe via an `endpoint` or `endpoints` attribute?)

### PR DESCRIPTION
**Pull request by radeksimko**
Fixes #2272 

I also removed all existing "exported" variables since these are actually input variables, exported _"by design"_. I don't think there's value in duplicating these.

If anyone thinks otherwise, then we should do it in docs of other resources as well, but I feel this would cause docs cluttering.

### CODE REVIEW

**Comment by catsby** [comment]
Is it stated elsewhere that input variables are exported by design? If so, :+1:

**Comment by radeksimko** [comment]
@catsby Not *all* input variable are always exported, but in 90% cases I'd guess they are. It's not mentioned anywhere since it's not 100%.

I can remove that single commit if you think it's better to keep it, but in that case we should change docs of other resources as well since these apparently have that assumption.

e.g. I have not seen `id` as exported variable mentioned anywhere explicitly, but it's used heavily in many examples.

TL;DR I don't mind doing it either way, I'm merely trying to keep docs consistent :smiley:

**Comment by phinze** [comment]
> Not all input variable are always exported, but in 90% cases I'd guess they are. It's not mentioned anywhere since it's not 100%.

Is this true? Based on the implementation of `Schema` I'd expect any config variable to always be available for referencing. Am I missing something, @radeksimko?

**Comment by radeksimko** [comment]
@phinze Now I'm confused since you made me think more deeply about it... :smile: (in a good way)

I thought it depends on whether we call `d.Set()` for those variables in `Read` methods, but I have not been checking how Terraform actually works on the HCL bridge with these.

I will double check once I get a moment.

**Comment by mitchellh** [comment]
I believe they're all available since due to merging, the config will be merged in, even if `Set` is not called.

**Comment by mitchellh** [comment]
But I agree its probably not worth repeating.... we'll see if this causes confusion...
